### PR TITLE
feat: allow public use

### LINKED ISSUE #112: Allow use while sign out

### PR DESCRIPTION
**Pull request by pyphilia**
In this PR I removed the forced login page.

Workflow:
- If signout, the app shows you the sign in screen. You can skip the login to show the app (but there's nothing to show for now). 
- If you're signed in you access the app directly

I removed the drawer as the profile is in the tab now.

close #112 

Screenshot 2023-11-16 at 14 25 56

simulator_screenshot_C656B0DC-CE9F-4F5C-B2D7-EDB2D48B90BF

simulator_screenshot_9350F84A-08BD-4F24-8CF1-EA09AC2E1DCF

simulator_screenshot_454E531E-2683-40BD-9118-0F8BA149FBEF

### CODE REVIEW

File: src/navigation/RootNavigator.tsx [hunk 1]
```diff
@@ -1,40 +1,31 @@
...
-} from './CommonStackNavigator';
-import DrawerNavigator, { DrawerParamList } from './DrawerNavigator';
+import CommonStackNavigator from './CommonStackNavigator';
+import MainStackNavigator from './MainStackNavigator';
 
 export type RootStackParamList = {
-  SignIn: { signUp: boolean; t?: string | undefined };
-  Drawer: NavigatorScreenParams<DrawerParamList>;
-  CommonStack: NavigatorScreenParams<CommonStackParamList>;
+  SignIn: any;
```
Comment (spaenleh) [inline src/navigation/RootNavigator.tsx:9]: Can we define a better type ?

File: src/navigation/TabNavigator.tsx [hunk 2]
```diff
@@ -28,26 +33,69 @@ const TabNavigator = () => {
...
+            tabPress: (e) => {
+              // Prevent default action
+              e.preventDefault();
+
+              // Do something with the `navigation` object
+              navigation.navigate('SignIn');
+            },
+          })}
+          options={{
+            tabBarLabel: t('Sign In')!,
```
Comment (spaenleh) [inline src/navigation/TabNavigator.tsx:91]: Is the `!` necessary ? the type should be able to infer that it wont be null. Maybe we need to set a fallback ?

File: src/screens/HomeScreen.tsx [hunk 3]
```diff
@@ -1,42 +1,57 @@
...
+export type HomeStackProps = CompositeScreenProps<
+  StackScreenProps<HomeStackParamList>,
+  CompositeScreenProps<
+    StackScreenProps<MainStackNavigatorParamList>,
+    StackScreenProps<HomeStackParamList>
+  >
 >;
 
-export type HomeStackPropsNavigationProp = HomeStackProps['navigation'];
+// export type HomeStackPropsNavigationProp = NavigationProp<HomeStackProps>;
```
Comment (spaenleh) [inline src/screens/HomeScreen.tsx:27]: Remove ?

**Review by spaenleh — APPROVED** [review]
Looks good to me ! 🐒 

I am having a bit of trouble understanding how all the stacks and navigators play together, maybe it would be nice to have a sketch of their interactions. This could also serve as a base to asses the user flows in the app.

File: src/navigation/TabNavigator.tsx (hunk 2 above)
Comment (pyphilia) [reply src/navigation/TabNavigator.tsx:91]: Upgrading the packages, done in the next PR.

**Pull request closed without merging**